web/files/file.go

```diff
@@ -87,6 +89,70 @@ func (f *FileDoc) ToJSONApi() ([]byte, error) {
 	return json.Marshal(m)
 }
 
+// GetFileDoc is used to fetch file document information form our
+// database.
+func GetFileDoc(fileID, dbPrefix string) (doc *FileDoc, err error) {
+	doc = &FileDoc{}
+	err = couchdb.GetDoc(dbPrefix, string(FileDocType), fileID, doc)
+	return
+}
+
+// ServeFileContent replies to a http request using the content of a
+// file given its FileDoc.
+//
+// It uses internally http.ServeContent and benefits from it by
+// offering support to Range, If-Modified-Since and If-None-Match
+// requests. It uses the revision of the file as the Etag value for
+// non-ranged requests
+//
+// The content disposition is inlined.
+func ServeFileContent(fileDoc *FileDoc, req *http.Request, w http.ResponseWriter, fs afero.Fs) (err error) {
+	attrs := fileDoc.Attrs
+	header := w.Header()
+	header.Set("Content-Type", attrs.Mime)
+	header.Set("Content-Disposition", "inline; filename="+attrs.Name+"")
+
+	if header.Get("Range") == "" {
+		header.Set("Etag", fileDoc.Rev())
+	}
+
+	serveContent(req, w, fs, fileDoc.Path, attrs.Name, attrs.UpdatedAt)
+	return
+}
+
+// ServeFileContentByPath replies to a http request using the content
+// of a file identified by its full path on the VFS. Unlike
+// ServeFileContent, this method does not require the full file
+// document but only its path.
+//
+// It also uses internally http.ServeContent but does not provide an
+// Etag.
```

Comment:
It's something that we will probably want to change later. Having Etag on this route would be very nice, because this route will be used to serve JS and CSS files for client-side apps and we want the browsers to avoid to download them each time the app loads.